definitions/npm/react-redux_v4.x.x/flow_v0.30.x-/react-redux_v4.x.x.js

```diff
@@ -53,7 +53,7 @@ type ConnectDispatchStateless = <P, SP, Dispatch: Function>(
   mapDispatchToProps: null | void,
   mergeProps: null | void,
   options?: {pure?: boolean, withRef?: boolean}
-) => (component: (props: P) => any) => Class<React$Component<void, $Diff<$Diff<P, {dispatch: Dispatch}>, SP>, void>>;
+) => (component: (props: P) => any) => Class<React$Component<void, $Diff<P, SP&{dispatch: Dispatch}>, void>>;
```

Comment:
I tried the Intersection type earlier but it didn't seem to work.

Have you tested this type with the newest Flow?

And when I say test, Flow will probably not throw an error on the type itself. But if you try to write some example code that depends on this code it should succeed for the correct types and catch your type errors.